packages/qwik-labs/src-vite/insights/index.ts

```diff
@@ -1,21 +1,52 @@
 import { type QwikVitePluginOptions } from '@builder.io/qwik/optimizer';
+import { existsSync } from 'fs';
+import { readFile, writeFile } from 'fs/promises';
+import { PluginOption } from 'vite';
 
-export async function insightsEntryStrategy({
-  publicApiKey,
-}: {
+const INSIGHTS_Q_MANIFEST_FILENAME = './dist/insights-q-manifest.json';
+
+const logWarn = (message?: any) => {
+  console.warn('\x1b[33m%s\x1b[0m', `\n\nQWIK WARN: ${message}\n`);
+};
+
+export async function qwikInsights(qwikInsightsOpts: {
   publicApiKey: string;
-}): Promise<QwikVitePluginOptions['entryStrategy']> {
-  const request = await fetch(`https://qwik-insights.builder.io/api/v1/${publicApiKey}/bundles/`);
-  try {
-    const bundles = await request.json();
-    return {
-      type: 'smart',
-      manual: bundles,
-    } as any;
-  } catch (e) {
-    console.error(e);
-    return {
-      type: 'smart',
-    };
-  }
+  baseUrl?: string;
+}): Promise<PluginOption> {
+  const { publicApiKey, baseUrl = 'https://qwik-insights.builder.io' } = qwikInsightsOpts;
+  let isProd = false;
+  const vitePlugin: PluginOption = {
+    name: 'vite-plugin-qwik-insights',
+    enforce: 'pre',
+    async config(viteConfig) {
+      isProd = viteConfig.mode !== 'ssr';
+      if (isProd) {
+        const qManifest: QwikVitePluginOptions['entryStrategy'] = { type: 'smart' };
+        try {
+          const response = await fetch(`${baseUrl}/api/v1/${publicApiKey}/bundles/`);
+          const bundles = await response.json();
+          qManifest.manual = bundles;
+        } catch (e) {
+          logWarn('fail to fetch manifest from Insights DB');
+        }
+        await writeFile(INSIGHTS_Q_MANIFEST_FILENAME, JSON.stringify(qManifest));
```

Comment:
Can we avoid the file creation and share the information through Vite plugins ( save somehow in memory ) ?
We need to pass a JSON object to `qwikVite` plugin https://github.com/BuilderIO/qwik/pull/5011/files#diff-60e4636a1ebc9915ae01bb9104aef297d0b3a1829e434a2e890c0fe06ab298eaR113

Reply:
I am fine with it being on disc, it may be useful for debugging.